client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -670,7 +671,20 @@ class CancelPurchaseForm extends Component {
 					<BlankCanvas className="cancel-purchase-form">
 						<BlankCanvas.Header onBackClick={ this.closeDialog }>
 							{ this.getHeaderTitle() }
-							<span className="cancel-purchase-form__site-slug">{ site.slug }</span>
+							<span className="cancel-purchase-form__site-slug">
+								{ ( () => {
+									switch ( purchase.productId ) {
+										// product.Id 5 is Domain Mapping
+										case 5:
```

Comment:
I don't know how the product data is created, but are we sure that these products will always keep these ids? I assume these are products that were injected as part of some data migration script, and that these ids will not change, but it's worth checking that, because this code might not work in a fresh instance of Calypso (though I'm not sure if this component is meant to be used outside of the WPCOM context).

Reply:
This was definitely a point of concern for me as well. Initially I'd used the product name, but that seemed to be too volatile. Any idea what team we could check with to confirm?

Reply:
I think you could check with @Automattic/shilling.

Reply:
`productId` should be stable and safe to use here. Another option is `productSlug`, neither of which are likely to change. For reference, `purchase` does have TS types which you can find [here](https://github.com/Automattic/wp-calypso/blob/6ebce469050693a803502fc84490e876a0c674e3/client/lib/purchases/types.ts#L53-L55).

Reply:
Thanks for confirming @sirbrillig , really appreciate the insight here! Since you've confirmed `productSlug` should be stable as well, I'll opt to use this, as it'll allow me to do away with the comments explaining what each `productId` is, as `productSlug` is self explanatory in each switch case.